### PR TITLE
Add authentication to websockets server

### PR DESCRIPTION
**Pull request by amCap1712**
The websockets will now run at https://listenbrainz.org/socket.io/ so that the websockets server can utilise Flask-Login cookies to authorize access to websocket rooms. Hence, only playlist creators and collaborator can join a playlist room.

### CODE REVIEW

File: listenbrainz/websockets/websockets.py [hunk 1]
```diff
@@ -42,10 +40,13 @@ def dispatch_playlist_updates(data):
 def joined(data):
     if 'playlist_id' not in data:
         raise BadRequest("Missing key 'playlist_id'")
-
-    room = data['playlist_id']
-    join_room(room)
-    emit('joined', {'status': 'success'}, to=room)
+    playlist_mbid = data['playlist_id']
+    playlist = db_playlist.get_by_mbid(playlist_mbid)
+    if current_user.is_authenticated and playlist.is_visible_by(current_user.id):
```
Comment (amCap1712) [inline listenbrainz/websockets/websockets.py:45]: Just checked, this should be `is_modifiable_by` instead. will fix.

**Review by MonkeyDo — APPROVED** [review]
Now that tests are passing, I think this is ready to merge

**Review by mayhem — APPROVED** [review]
:shinto_shrine:

**Comment by amCap1712** [comment]
Thanks @MonkeyDo and @mayhem

**Pull request merged**